src/Microsoft.Diagnostics.EventFlow.Inputs.ILogger/Logger/EventFlowLogger.cs

```diff
@@ -0,0 +1,111 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.ILogger.Logger
+{
+    internal class EventFlowLogger : Extensions.Logging.ILogger
+    {
+        private readonly string categoryName;
+        private readonly LoggerInput loggerInput;
+        private Scope<object> scope;
+
+        public EventFlowLogger(string categoryName, LoggerInput loggerInput)
+        {
+            Debug.Assert(categoryName != null);
+            Debug.Assert(loggerInput != null);
+
+            this.categoryName = categoryName;
+            this.loggerInput = loggerInput;
+        }
+
+        public void Log<TState>(Extensions.Logging.LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+                return;
+            var message = formatter(state, exception);
+
+            Dictionary<string, object> properties = new Dictionary<string, object>();
+            if (state is FormattedLogValues)
+            {
+                var formattedState = state as FormattedLogValues;
+                //last KV is the whole message, we will pass it separately
+                for (int i = 0; i < formattedState.Count - 1; i++)
+                    properties.Add(formattedState[i].Key, formattedState[i].Value);
+            }
+            if (scope?.State != null)
+            {
+                var formattedState = scope.State as FormattedLogValues;
+                if (formattedState != null)
+                {
+                    for (int i = 0; i < formattedState.Count - 1; i++)
+                    {
+                        if (!properties.ContainsKey(formattedState[i].Key)) //message and scope may have the same property
+                            properties.Add(formattedState[i].Key, formattedState[i].Value);
+                    }
+                    //last KV is the whole 'scope' message, we will add it formatted
+                    properties.Add("Scope", formattedState.ToString());
```

Comment:
We can check, but what to do with duplicate? 
I see two possible ways
1. Ignore duplicates and report error to health reporter.
2. Keep all properties from the message in separate dict under payload, but then LoggerInput would behave differently than EventSource or Serilog. 


EventSourceInput and Serilog seems to have the same problem. So may be all of them should be fixed in the same way?